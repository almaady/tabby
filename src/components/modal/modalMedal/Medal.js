import React from "react";
import PropTypes from "prop-types";
/**
 * @Class Medal - The image of the medal
 */
class Medal extends React.PureComponent {
  /**
   * @property {string} medalSrc - Medal URL
   */
  static propTypes = {
    medalSrc: PropTypes.string.isRequired
  };

  render() {
    const { medalSrc } = this.props;
    return <img alt="Medal" src={medalSrc} />;
  }
}

export default Medal;
