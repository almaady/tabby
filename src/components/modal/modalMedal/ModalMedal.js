// Modal Medal
import React from "react";
import PropTypes from "prop-types";

import Modal from "../Modal";
import ModalHeader from "../ModalHeader";
import ModalFooter from "../ModalFooter";
import ButtonText from "../../buttons/ButtonText";

/**
 * @Class Modal Medal - Is the modal for the Medals, it has its own style, but shares the Modal component
 */
class ModalMedal extends React.PureComponent {
  /**
   * @property {bool} show - Determines if the modal is visible
   * @property {func} onClose - The function to close the modal
   * @property {node} children - Medals, could be more than one
   * @property {string} title - The title of the medal
   * @property {string} message - The message of the modal
   * @property {string} btnText - The text of the button
   * @property {string} btnColor - The color of the button, it's blue by default
   * @see See './../../../styles/variables/_colors.scss' for the list of color names
   */
  static propTypes = {
    show: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    title: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    btnText: PropTypes.string.isRequired,
    btnColor: PropTypes.string
  };

  render() {
    const {
      show,
      onClose,
      children,
      title,
      btnText,
      btnColor,
      message
    } = this.props;
    return (
      <Modal show={show} className="modal--medal" onClose={onClose}>
        <div className="medals">
          {React.Children.map(children, child => child)}
        </div>
        {/** Starts stars animation */}
        <div className="stars--container">
          <div id="stars" />
          <div id="stars2" />
          <div id="stars3" />
        </div>
        {/** Ends stars animation */}
        <ModalHeader title={title} />
        <p>{message}</p>
        <ModalFooter>
          <ButtonText text={btnText} color={btnColor} onClick={onClose} />
        </ModalFooter>
      </Modal>
    );
  }
}
ModalMedal.defaultProps = {
  btnColor: "blue"
};
export default ModalMedal;
