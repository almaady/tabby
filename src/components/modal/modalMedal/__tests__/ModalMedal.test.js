import React from "react";
import { shallow } from "enzyme";

import sinon from "sinon";
import ModalMedal from "../ModalMedal";
import Medal from "../Medal";

describe("Modal Medal", () => {
  const close = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      show: false,
      onClose: close,
      children: <div />,
      title: "Title",
      message: "message",
      btnText: "button text"
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(
      <ModalMedal {...props}>
        <Medal medalSrc="medal" />
        <Medal medalSrc="medal" />
        <Medal medalSrc="medal" />
      </ModalMedal>
    );
    expect(component).toMatchSnapshot();
  });
  it("should remove hiddden class when show is true", () => {
    const propsVisible = { ...props };
    propsVisible.show = true;
    const componentVisible = shallow(<ModalMedal {...propsVisible} />);
    expect(componentVisible.hasClass("hidden")).toEqual(false);
  });
  it("should call close when you click on ButtonText", () => {
    const component = shallow(<ModalMedal {...props} />);
    component.find("ButtonText").simulate("click");
    expect(close.called).toBeTruthy();
  });
});
