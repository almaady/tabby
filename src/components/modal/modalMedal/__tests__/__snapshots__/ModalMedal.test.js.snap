// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Medal should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalMedal
    btnColor="blue"
    btnText="button text"
    message="message"
    onClose={[Function]}
    show={false}
    title="Title"
  >
    <Medal
      medalSrc="medal"
    />
    <Medal
      medalSrc="medal"
    />
    <Medal
      medalSrc="medal"
    />
  </ModalMedal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="medals"
        >
          <Medal
            medalSrc="medal"
          />
          <Medal
            medalSrc="medal"
          />
          <Medal
            medalSrc="medal"
          />
        </div>,
        <div
          className="stars--container"
        >
          <div
            id="stars"
          />
          <div
            id="stars2"
          />
          <div
            id="stars3"
          />
        </div>,
        <ModalHeader
          title="Title"
        />,
        <p>
          message
        </p>,
        <ModalFooter>
          <ButtonText
            borderColor=""
            className=""
            color="blue"
            disabled={false}
            onClick={[Function]}
            text="button text"
            type="button"
          />
        </ModalFooter>,
      ],
      "className": "modal--medal",
      "classNameWrapper": "",
      "onClose": [Function],
      "show": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Medal
              medalSrc="medal"
            />,
            <Medal
              medalSrc="medal"
            />,
            <Medal
              medalSrc="medal"
            />,
          ],
          "className": "medals",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": ".0",
            "nodeType": "class",
            "props": Object {
              "medalSrc": "medal",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".1",
            "nodeType": "class",
            "props": Object {
              "medalSrc": "medal",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".2",
            "nodeType": "class",
            "props": Object {
              "medalSrc": "medal",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="stars"
            />,
            <div
              id="stars2"
            />,
            <div
              id="stars3"
            />,
          ],
          "className": "stars--container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "stars",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "stars2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "stars3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "message",
        },
        "ref": null,
        "rendered": "message",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ButtonText
            borderColor=""
            className=""
            color="blue"
            disabled={false}
            onClick={[Function]}
            text="button text"
            type="button"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "borderColor": "",
            "className": "",
            "color": "blue",
            "disabled": false,
            "onClick": [Function],
            "text": "button text",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="medals"
          >
            <Medal
              medalSrc="medal"
            />
            <Medal
              medalSrc="medal"
            />
            <Medal
              medalSrc="medal"
            />
          </div>,
          <div
            className="stars--container"
          >
            <div
              id="stars"
            />
            <div
              id="stars2"
            />
            <div
              id="stars3"
            />
          </div>,
          <ModalHeader
            title="Title"
          />,
          <p>
            message
          </p>,
          <ModalFooter>
            <ButtonText
              borderColor=""
              className=""
              color="blue"
              disabled={false}
              onClick={[Function]}
              text="button text"
              type="button"
            />
          </ModalFooter>,
        ],
        "className": "modal--medal",
        "classNameWrapper": "",
        "onClose": [Function],
        "show": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Medal
                medalSrc="medal"
              />,
              <Medal
                medalSrc="medal"
              />,
              <Medal
                medalSrc="medal"
              />,
            ],
            "className": "medals",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": ".0",
              "nodeType": "class",
              "props": Object {
                "medalSrc": "medal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": ".1",
              "nodeType": "class",
              "props": Object {
                "medalSrc": "medal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": ".2",
              "nodeType": "class",
              "props": Object {
                "medalSrc": "medal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="stars"
              />,
              <div
                id="stars2"
              />,
              <div
                id="stars3"
              />,
            ],
            "className": "stars--container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "stars",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "stars2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "stars3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "message",
          },
          "ref": null,
          "rendered": "message",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ButtonText
              borderColor=""
              className=""
              color="blue"
              disabled={false}
              onClick={[Function]}
              text="button text"
              type="button"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "borderColor": "",
              "className": "",
              "color": "blue",
              "disabled": false,
              "onClick": [Function],
              "text": "button text",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
